### DISCUSSION TITLE
How to identify incorrect configuration in Kong proxy?

### DISCUSSION THREAD

**Discussion by surenraju-careem** [Kong/kong#12902] [Help]
Hello Kong Community,

We are running Kong Ingress Controller version 2.11 with Kong Proxy version 3.3 in DBless mode. Due to spiked memory usage in the Kong Ingress Controller container, the KIC container was restarted. Fourteen out of fifteen pods automatically recovered. However, one of the KIC containers and the Kong Ingress container did not have any routes for any of the Ingresses, and this pod became healthy. It rejected all traffic with a 404 response.

From the Kong Proxy side, is there any way to monitor the number of services and routes configured so that we can monitor such scenario and alert? AFAIK the prometheus metrics does not provide such information.

**Comment by StarlightIbuki** [comment]
Currently, it seems we do not have a perfect user-facing interface for this, and we can do it by querying the DB directly.
IMO, this could be helpful if we include it as a part of `admin_api`. @ADD-SP .

**Reply by ADD-SP** [reply]
> we can do it by querying the DB directly

KIC uses DB-less mode, there is no backend database.

**Reply by StarlightIbuki** [reply]
For KIC we can export the config from nodes and then count the entities with `jq`.
